gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/service/impl/SubscriptionServiceImpl.java

```diff
@@ -509,6 +509,17 @@ public SubscriptionEntity process(ProcessSubscriptionEntity processSubscription,
                 subscription.setClosedAt(new Date());
             }
 
+            if (planEntity.getSecurity() == PlanSecurityType.API_KEY && subscription.getStatus() == Subscription.Status.ACCEPTED) {
+                if (StringUtils.isNotEmpty(processSubscription.getCustomApiKey())) {
+                    if (apiKeyService.exists(processSubscription.getCustomApiKey())) {
```

Comment:
I wonder if we should also add this check in `ApiServiceImpl.renew`.
As we check it in the UI, in the 'renew' popup, but not in backoffice side (calling the renew endpoint with a already used custom API key leads to a DuplicateKeyException)

Reply:
Maybe the best place to could be the [generateForSubscription](https://github.com/gravitee-io/gravitee-api-management/blob/3.10.x/gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/service/impl/ApiKeyServiceImpl.java#L198) method as it's called by both renew and approval of the subscription but I'll need to retest everything linked to this method or the parents.. 😅